doi/src/main/java/ca/nrc/cadc/doi/DoiInitAction.java

```diff
@@ -251,12 +264,76 @@ private static MultiValuedProperties getConfig(boolean verify) {
                 sb.append("OK");
             }
         }
-        log.info(sb.toString());
 
         if (!ok) {
-            throw new IllegalStateException("incomplete config: " + sb.toString());
+            throw new IllegalStateException("incomplete config: " + sb);
         }
         return props;
     }
 
+    // check that the DOI parent node path, configured with the V0SPACE_PARENT_PATH_KEY property,
+    // exists and has the expected properties.
+    private static void checkParentFolders() {
+        MultiValuedProperties config = getConfig();
+        URI vospaceResourceID = URI.create(config.getFirstPropertyValue(DoiInitAction.VAULT_RESOURCE_ID_KEY));
+        String parentPath = config.getFirstPropertyValue(PARENT_PATH_KEY);
+
+        Subject adminSubject = SSLUtil.createSubject(new File("/config/doiadmin.pem"));
+        adminSubject = AuthenticationUtil.augmentSubject(adminSubject);
+        String adminUsername = getUsername(adminSubject);
+
+        VOSpaceClient vosClient = new VOSpaceClient(vospaceResourceID);
+        String currentPath = "";
+        String[] paths = parentPath.split("/");
+        for (String path : paths) {
```

Comment:
Is this recursive checking required? It is performed on the server side as well and reported back to the client.

Reply:
That's a good question. We use two container nodes as the root for the DOI's: **/AstroDataCitationDOI/CISTI.CANFAR**, but the owner and permissions only matter on the second node. I do remember something about why two parent nodes, and why owner and permissions matter for both, but I can't find any reference. Tempted to change that check for only the second parent node, but I'll leave it until I'm more sure. Added a TODO there so I don't forget.